src/dsl/validation.rs

```diff
@@ -27,7 +30,42 @@ pub fn validate(
         }
     };
 
-    validation_result
+    let mut schema_validation_errors = match validation_result {
+        Ok(_) => vec![],
+        Err(errors) => errors,
+    };
+
+    let (_, expression_errors): (Vec<_>, Vec<_>) = json_check
+        .get("expectations")
+        .unwrap_or(&json!([]))
+        .as_array()
+        .unwrap_or(&Vec::new())
+        .iter()
+        .map(|value| {
+            let expect = value.get("expect").unwrap().as_str().unwrap();
```

Comment:
We are missing the `expect_same` scenario here.
We should get one or the other.

Reply:
Right :+1: